ConwaysGameOfLife/Generator.cs

```diff
@@ -28,13 +32,39 @@ public Generator()
 
         */
 
-        public bool ShouldSwitchCellState()
+        public Universe GenerateNewUniverse()
+        {
+            var oldCells = _oldUniverse.Cells;
+            var newCells = new List<Cell>();
+            foreach (var cell in oldCells)
+            {
+                GenerateCells(newCells, cell);
+            }
+            return new Universe(_oldUniverse.GridWidth, _oldUniverse.GridLength, newCells);
+        }
+
+        private void GenerateCells(List<Cell> cells, Cell cell)
         {
-            throw new NotImplementedException();
+            var newCell = GenerateNewCell(cell);
+            cells.Add(newCell);
         }
-        public Universe GenerateNewUniverse(Universe universe)
+
+        private Cell GenerateNewCell(Cell cell)
         {
-            throw new NotImplementedException();
+            var numberOfLiveNeighbours = _oldUniverse.CountLiveNeighbours(cell);
+            var state = cell.CellState;
+            var newCell = new Cell(state, cell.Location.X, cell.Location.Y);
+            ShouldSwitchCellState(numberOfLiveNeighbours, state, newCell);

```

Comment:
Does calling `ShouldSwitchCellState` here make sense? What if the cell doesn't need to switch cell state?
Method name indicates it should return a boolean.
If it should switch cell state, set the new state in this function.

Reply:
```suggestion
            if (ShouldSwitchCellState(numberOfLiveNeighbours, state)) newCell.SwitchCellState();
```

see next comment alongside this one